### ISSUE TITLE
Crash report from Dynamo 2.3.0.5123

### ISSUE THREAD

**Issue by vykrum** [DynamoDS/Dynamo#9764]
# Issue Description
Please fill in the following information to help us reproduce the issue:

### What did you do?
Increased value in defined function

[20190602-1.zip](https://github.com/DynamoDS/Dynamo/files/3247925/20190602-1.zip)

### What did you expect to see?
Longer list
![img1](https://user-images.githubusercontent.com/5621624/58807792-62798d80-8636-11e9-9d60-710aafbc65cd.png)

### What did you see instead?
![img2](https://user-images.githubusercontent.com/5621624/58807832-72916d00-8636-11e9-9be6-2c6fa0f97838.png)

### What packages or external references (if any) were used?
None

### Stack Trace
```
Object reference not set to an instance of an object.

   at ProtoCore.DSASM.Heap.RecursiveMark(StackValue root)
   at ProtoCore.DSASM.Heap.SingleStep(Boolean forceGC)
   at ProtoCore.DSASM.Heap.FullGC(IEnumerable`1 gcroots, Executive exe)
   at ProtoScript.Runners.LiveRunner.ApplyUpdate()
   at ProtoScript.Runners.LiveRunner.CompileAndExecuteForDeltaExecution(List`1 astList)
   at ProtoScript.Runners.LiveRunner.SynchronizeInternal(GraphSyncData syncData)
   at ProtoScript.Runners.LiveRunner.UpdateGraph(GraphSyncData syncData)
   at Dynamo.Scheduler.UpdateGraphAsyncTask.HandleTaskExecutionCore()
   at Dynamo.Scheduler.AsyncTask.Execute()
```

---
OS: `Microsoft Windows NT 6.2.9200.0`
CLR: `4.0.30319.42000`
Dynamo: `2.3.0.5123`

**Comment by vykrum** [comment]
```
Object reference not set to an instance of an object.

   at ProtoCore.Utils.StringUtils.ConcatString(StackValue op1, StackValue op2, RuntimeCore runtimeCore)
   at ProtoCore.Utils.CoreUtils.AddStackValueString(StackValue sv1, StackValue sv2, RuntimeCore runtimeCore)
   at ProtoCore.DSASM.Executive.ADD_Handler(Instruction instruction)
   at ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   at ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   at ProtoCore.DSASM.Interpreter.Run(Int32 codeblock, Int32 entry, Language lang, List`1 breakpoints)
   at ProtoCore.Lang.JILFunctionEndPoint.Execute(Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore)
   at ProtoCore.CallSite.ExecWithZeroRI(List`1 functionEndPoint, Context c, List`1 formalParameters, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData, FunctionEndPoint finalFep)
   at ProtoCore.CallSite.ExecWithRISlowPath(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData, FunctionEndPoint finalFunctionEndPoint)
   at ProtoCore.CallSite.ExecWithRISlowPath(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore, SingleRunTraceData previousTraceData, SingleRunTraceData newTraceData, FunctionEndPoint finalFunctionEndPoint)
   at ProtoCore.CallSite.Execute(List`1 functionEndPoint, Context c, List`1 formalParameters, List`1 replicationInstructions, StackFrame stackFrame, RuntimeCore runtimeCore)
   at ProtoCore.CallSite.DispatchNew(Context context, List`1 arguments, List`1 partialReplicationGuides, DominantListStructure domintListStructure, StackFrame stackFrame, RuntimeCore runtimeCore)
   at ProtoCore.DSASM.Executive.Callr(Int32 blockDeclId, Int32 functionIndex, Int32 classIndex, Boolean& explicitCall, Boolean isDynamicCall, Boolean hasDebugInfo)
   at ProtoCore.DSASM.Executive.CALLR_Handler(Instruction instruction)
   at ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, Language language)
   at ProtoCore.DSASM.Executive.Execute(Int32 exeblock, Int32 entry, List`1 breakpoints, Language language)
   at ProtoCore.DSASM.Executive.BounceUsingExecutive(Executive executive, Int32 exeblock, Int32 entry, StackFrame stackFrame, Int32 locals, Boolean fepRun, Executive exec, List`1 breakpoints)
   at ProtoScript.Runners.ProtoScriptRunner.ExecuteLive(Core core, RuntimeCore runtimeCore)
   at ProtoScript.Runners.LiveRunner.Execute(Boolean isCodeCompiled)
   at ProtoScript.Runners.LiveRunner.CompileAndExecute(List`1 astList)
   at ProtoScript.Runners.LiveRunner.CompileAndExecuteForDeltaExecution(List`1 astList)
   at ProtoScript.Runners.LiveRunner.SynchronizeInternal(GraphSyncData syncData)
   at ProtoScript.Runners.LiveRunner.UpdateGraph(GraphSyncData syncData)
   at Dynamo.Scheduler.UpdateGraphAsyncTask.HandleTaskExecutionCore()
   at Dynamo.Scheduler.AsyncTask.Execute()
```

**Comment by mjkkirschner** [comment]
@aparajit-pratap is this the same issue as:
https://jira.autodesk.com/browse/DYN-1637

**Comment by mjkkirschner** [comment]
tracked with https://jira.autodesk.com/browse/DYN-2291

**Comment by avidit** [comment]
Fixed in 2.13
